### DISCUSSION TITLE
gRPC Support

### DISCUSSION THREAD

**Discussion by VladislavYakonyuk** [mnfst/llm-gateway#213] [Feature request]
Hi, thank you so much for the tool idea presented, we will definitely try it.

We also have an idea to make gRPC support. To genirize a moc-server based on proto-files. We have been reserching in this direction, but didn't find anything sensible and working

**Comment by brunobuddy** [comment]
Hello @VladislavYakonyuk,

Thank you for your message, I am not familiar with gRPC. I see that the objective is to call server functions in the client like they are local correct ? We have a [JS SDK](https://manifest.build/docs/javascript-sdk) that does a bit like that.

We also have users that requested to add [tRPC](https://trpc.io/) for type safety. Are you looking for something similar ?

**Reply by joeberetta** [reply]
Hi @brunobuddy grpc differs from rest api in 2 points:
- binary format (it's more efficient than rest's more declarative and json format)
- more speedy (cuz binary and compressed)

It's dirty and fast explanation

gRPC is not directy about rpc as remote procedure call, as it would looks like)

Hope it was helpful

**Comment by brunobuddy** [comment]
Closing this out. It's from Manifest's old backend-as-a-service days, which we don't work on anymore since the pivot. See #742 for where things are now.
